src-java/kilda-model/src/main/java/org/openkilda/model/FlowPath.java

```diff
@@ -359,6 +371,33 @@ public void setSegments(List<PathSegment> segments) {
 
             this.segments = new ArrayList<>(segments);
         }
+
+
+        @Override
+        public void addFlowMirrorPoints(FlowMirrorPoints flowMirrorPoints) {
+            boolean toBeAdded = true;
+            Iterator<FlowMirrorPoints> it = this.flowMirrorPointsSet.iterator();
+            while (it.hasNext()) {
```

Comment:
Looks like `Map` (with some switchid+groupid immutable class as key) is more suitable here that `Set`.

PS And why we have filter on switchid+groupid instead of simple switch-id - as far as I understood path object can have only one mirror point per switch (because it represent one direction)?